src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.ByteArray.cs

```diff
@@ -110,34 +111,33 @@ public static byte[] SerializeToUtf8Bytes(object? value, Type inputType, JsonSer
                 ThrowHelper.ThrowArgumentNullException(nameof(context));
             }
 
-            Type runtimeType = GetRuntimeTypeAndValidateInputType(value, inputType);
-            JsonTypeInfo jsonTypeInfo = GetTypeInfo(context, runtimeType);
-            return WriteBytesUsingGeneratedSerializer(value!, jsonTypeInfo);
+            ValidateInputType(value, inputType);
+            JsonTypeInfo jsonTypeInfo = GetTypeInfo(context, inputType);
+            return WriteBytesAsObject(value, jsonTypeInfo);
         }
 
-        private static byte[] WriteBytesUsingGeneratedSerializer<TValue>(in TValue value, JsonTypeInfo jsonTypeInfo)
+        private static byte[] WriteBytes<TValue>(in TValue value, JsonTypeInfo<TValue> jsonTypeInfo)
         {
+            Debug.Assert(jsonTypeInfo?.IsConfigured == true);
+
             JsonSerializerOptions options = jsonTypeInfo.Options;
 
             using var output = new PooledByteBufferWriter(options.DefaultBufferSize);
-            using (var writer = new Utf8JsonWriter(output, options.GetWriterOptions()))
-            {
-                WriteUsingGeneratedSerializer(writer, value, jsonTypeInfo);
-            }
+            using var writer = new Utf8JsonWriter(output, options.GetWriterOptions());
```

Comment:
actually previous logic made more sense to me because we disposed of the writer earlier so it was clear that `output.WrittenMemory.ToArray()` converted bytes. With this specific usage I expect in some cases we might not convert something (i.e. closing brace etc.)

Reply:
The `WriteCore` methods all do flushing of their own, so even though the sequence of disposal changes somewhat, it only concerns clearing state and returning buffers without any real functional impact.